lib/stream/utils.js

```diff
@@ -0,0 +1,42 @@
+const errors = require('../errors');
+
+/**
+ * read a JSON object from a stream returned as a javascript object,
+ * handle errors.
+ *
+ * @param {stream.Readable} s - Readable stream
+ * @param {@hapi/joi} [joiSchema] - optional validation schema for the JSON object
+ * @return {Promise} a Promise resolved with the parsed JSON object as a result
+ */
+async function readJSONStreamObject(s, joiSchema) {
+    return new Promise((resolve, reject) => {
+        const contentsChunks = [];
+        s.on('data', chunk => {
+            contentsChunks.push(chunk);
+        });
+        s.on('end', () => {
+            const contents = contentsChunks.join('');
+            try {
+                const parsedContents = JSON.parse(contents);
+                let validatedContents;
```

Comment:
Could just resolve(value) and resolve(parsedContents) instead of creating this variable. Doesn't matter though, either way.